test/unit/world-name-permission-checker.spec.ts

```diff
@@ -0,0 +1,209 @@
+    })
+
+    await expect(permissionChecker.checkPermission('0xb', name)).resolves.toBeFalsy()
+  })
+
+  it('when on chain validation returns false', async () => {
+    const permissionChecker = await createOnChainDclNameChecker({
+      config,
+      logs,
+      ethereumProvider: createHttpProviderMock({
+        jsonrpc: '2.0',
+        id: 1,
+        result: '0x0000000000000000000000000000000000000000000000000000000000000000'
+      })
+    })
+
+    const identity = await getIdentity()
+    const address = identity.authChain.authChain[0].payload
+    await expect(permissionChecker.checkPermission(address, 'my-super-name.dcl.eth')).resolves.toBeFalsy()
+  })
+
+  it('when on chain validation returns true', async () => {
+    const permissionChecker = await createOnChainDclNameChecker({
+      config,
+      logs,
+      ethereumProvider: createHttpProviderMock({
+        jsonrpc: '2.0',
+        id: 1,
+        result: '0x0000000000000000000000000000000000000000000000000000000000000001'
+      })
+    })
+
+    const identity = await getIdentity()
+    const address = identity.authChain.authChain[0].payload
+    await expect(permissionChecker.checkPermission(address, 'my-super-name.dcl.eth')).resolves.toBeTruthy()
+  })
+})
+
+describe('name checker: endpoint', function () {
+  let logs: ILoggerComponent
+  let config: IConfigComponent
+  let fetch: IFetchComponent
+
+  beforeEach(async () => {
+    config = createConfigComponent({
+      LOG_LEVEL: 'DEBUG',
+      ENDPOINT_NAME_CHECKER_BASE_URL: 'http://anything'
+    })
+    logs = await createLogComponent({ config })
+    fetch = {
+      fetch: async (_url: Request): Promise<Response> => new Response(undefined)
+    }
+  })
+
+  it('when permission asked for invalid name returns false', async () => {
+    const permissionChecker = await createEndpointNameChecker({
+      config,
+      fetch,
+      logs
+    })
+
+    await expect(permissionChecker.checkPermission('0xb', '')).resolves.toBeFalsy()
+  })
+
+  it('when permission asked for invalid address returns false', async () => {
+    const permissionChecker = await createEndpointNameChecker({
+      config,
+      fetch,
+      logs
+    })
+
+    await expect(permissionChecker.checkPermission('', 'anything')).resolves.toBeFalsy()
+  })
+
+  it.each([true, false])('when valid name and address it returns as per the endpoint', async (value) => {
+    fetch = {
+      fetch: async (_url: Request): Promise<Response> => new Response(String(value))
+    }
+
+    const permissionChecker = await createEndpointNameChecker({
+      config,
+      fetch,
+      logs
+    })
+
+    const identity = await getIdentity()
+    const address = identity.authChain.authChain[0].payload
+    await expect(permissionChecker.checkPermission(address, 'my-super-name.dcl.eth')).resolves.toBe(value)
+  })
+})
+
+describe('name checker: noop', function () {
+  let logs: ILoggerComponent
+  let config: IConfigComponent
```

Comment:
We don't need these to build NoOp Checker, right?

Reply:
Good catch! Removing.